daisy_workflows/image_build/debian/build_fai.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Build the Debian image on a GCE VM.
+
+Parameters (retrieved from instance metadata):
+
+fai_cloud_images_version: The debian's fai-cloud-images scripts git commit ID
+to use.
+debian_version: The FAI tool debian version to be requested.
+image_dest: The Cloud Storage destination for the resultant image.
+"""
+
+import collections
+import json
+import logging
+import os
+import tarfile
+import urllib
+import zipfile
+
+import utils
+
+utils.AptGetInstall(['qemu-utils', 'fai-server', 'fai-setup-storage'])
+
+
+def main():
+  # Get Parameters.
+  fai_cloud_images_version = utils.GetMetadataAttribute(
+      'fai_cloud_images_version', raise_on_not_found=True)
+  debian_version = utils.GetMetadataAttribute(
+      'debian_version', raise_on_not_found=True)
+  image_dest = utils.GetMetadataAttribute('image_dest',
+      raise_on_not_found=True)
+  outs_path = utils.GetMetadataAttribute('daisy-outs-path',
+      raise_on_not_found=True)
+
+  logging.info('fai-cloud-images version: %s' % fai_cloud_images_version)
+  logging.info('debian version: %s' % debian_version)
+
+  # First, update fai-client as the older version on stretch will not work
```

Comment:
I think there was an upstream version in a deb repo here: https://fai-project.org/download/

Reply:
oh, look at that! Great idea! I'll use that